src/services/util.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Helper function to toggle password field visibility
+ *
+ * @export
+ * @param {Event} e
+ * @param {HTMLInputElement} passwordField
+ */
+export async function togglePassword(
+  e: Event,
+  passwordField: HTMLInputElement
+) {
+  const button = e.target as HTMLImageElement;
+  const field = passwordField;
+
+  if (field?.type === 'password') {
+    field.type = 'text';
+    button.src = 'https://archive.org/images/eye.svg';
+    button.alt = 'View text';
+  } else {
+    field.type = 'password';
+    button.src = 'https://archive.org/images/eye-crossed.svg';
+    button.alt = 'Hide text';
+  }
+}
+
+/**
+ * Helper function to stop page refresh
+ *
+ * @export
+ * @param {Event} e
+ */
+export async function preventDefault(e: Event) {
+  e?.preventDefault();
+  e?.stopPropagation();
+}
```

Comment:
Suggestion: perhaps this function name could more accurately reflect that it does more than just `preventDefault`, to avoid confusion. Also, no need for this to be `async`.